replica/database.cc

```diff
@@ -929,11 +930,10 @@ future<> database::drop_column_family(const sstring& ks_name, const sstring& cf_
     dblog.debug("Dropping {}.{}", ks_name, cf_name);
     co_await remove(*cf);
     cf->clear_views();
-    co_return co_await cf->await_pending_ops().then([this, &ks, cf, tsf = std::move(tsf), snapshot] {
-        return truncate(ks, *cf, std::move(tsf), snapshot).finally([this, cf] {
-            return cf->stop();
-        });
-    }).finally([cf] {});
+    co_await cf->await_pending_ops();
+    auto f = co_await coroutine::as_future(truncate(ks, *cf, std::move(tsf), snapshot));
+    co_await cf->stop();
```

Comment:
This will back backports harder, but I don't mind.

Reply:
We only have to backport the last patch, the rest is cosmetic.